src/chainstate/coordinator/mod.rs

```diff
@@ -658,6 +708,281 @@ impl<
+                        .expect_optional();
+
+                    let num_vetos = match entry {
+                        Some(val) => {
+                            let tuple_data = val.expect_tuple();
+                            tuple_data
+                                .get("vetoes")
+                                .expect("BUG: malformed cost proposal tuple")
+                                .to_owned()
+                                .expect_u128()
+                        }
+                        None => 0,
+                    };
+
+                    info!(
+                        "chains coordinator: in veto check, proposed_exit: {}, num vetos: {:?}",
+                        proposed_exit_rc, num_vetos
+                    );
+                    // Check if the percent veto crosses the minimum threshold
+                    let percent_veto = num_vetos * 100 / (reward_cycle_length as u128);
+
+                    Ok(percent_veto >= (veto_percent_threshold as u128))
+                })
+            })
+            .ok_or(Error::DBError(NotFoundError))?
+    }
+
+    /// Returns map of exit proposals to the number of votes for each, as well as the sum total of all
+    /// votes.
+    /// The range of this map corresponds to rc_cycle_of_vote + EXIT_RC_MINIMUM_RC_BUFFER_FROM_PRESENT
+    /// to rc_cycle_of_vote + EXIT_RC_MAXIMUM_RC_BUFFER_FROM_PRESENT.
+    pub fn read_vote_state(
+        &mut self,
+        rc_cycle_of_vote: u64,
+        curr_exit_at_rc_opt: Option<u64>,
+        consensus_hash: &ConsensusHash,
+        block_hash: &BlockHeaderHash,
+        invalid_reward_cycles: Vec<u64>,
+        exit_contract_id: &QualifiedContractIdentifier,
+    ) -> Result<(BTreeMap<u64, u128>, u128), Error> {
+        let stacks_block_id = StacksBlockHeader::make_index_block_hash(consensus_hash, block_hash);
+        let mut vote_map = BTreeMap::new();
+        let mut min_rc = self
+            .burnchain
+            .exit_contract_constants
+            .absolute_minimum_exit_rc
+            .max(
+                rc_cycle_of_vote
+                    + self
+                        .burnchain
+                        .exit_contract_constants
+                        .minimum_rc_buffer_from_present,
+            );
+        let max_rc = rc_cycle_of_vote
+            + self
+                .burnchain
+                .exit_contract_constants
+                .maximum_rc_buffer_from_present;
+
+        // Check what value is stored for the current exit at rc.
+        // If there is an existing exit rc, make sure the minimum rc we consider for the votes is
+        // greater than it.
+        if let Some(curr_exit_at_rc) = curr_exit_at_rc_opt {
+            min_rc = min_rc.max(curr_exit_at_rc + 1);
+        }
+
+        let mut total_votes = 0;
+        let invalid_reward_cycles: HashSet<u64> =
+            HashSet::from_iter(invalid_reward_cycles.into_iter());
+        let is_mainnet = self.burnchain.is_mainnet();
```

Comment:
This method is broken, but we can't fix it because it's consensus-critical.  `Burnchain::is_mainnet()` always returns `false`, since it compares the Stacks network ID to the Bitcoin network ID (oops).

Reply:
Can you add a docstring to `Burnchain::is_mainnet()` to say something to the effect of `"BROKEN! DO NOT USE IN NEW PLACES!"

Reply:
Should I make a new is_mainnet_function (like `is_mainnet_v2`) and use that instead. In that function I would compare the Stacks network ID against `BITCOIN_NETWORK_ID_MAINNET`.

Reply:
It doesn't look like this value is ever used in this method -- I think just remove this variable.